pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,412 @@
+package jobruntestcaseanalyzer
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+type testIdentifier struct {
+	testSuites []string
```

Comment:
What is the scenario in which we might be looking for a test amidst multiple suites? Is this anticipating openshift-tests vs openshift-tests-upgrade?

Reply:
Test suite can be embedded. This design allows one to match internal test suites.